app/assets/javascripts/discourse/components/composer-editor.js.es6

```diff
@@ -862,6 +862,64 @@ export default Ember.Component.extend({
     }
   },
 
+  _placeImageScaleButtons($preview) {
+    const images = $preview.find("img.upload-img");
+
+    const regexString = `(\\!\\[\\S[^\\]]+\\|\\d{3,6}x\\d{3,6})(?:,(\\d+)\\%)?([^\\]]*?\\]\\(upload\\:\\/\\/(?:.)*?\\))`;
```

Comment:
The double escape is making this regexp really hard to read. Why aren't you declaring the regex directly?

```js
const uploadRegexp = /(!\[\S[^\]]+|\d+x\d+)(?::,(\d+)%)?([^\]]*?\]\(update:\/\/(?:.)*?\)))/g;
```